app/services/geo_resources/discovery/document_builder/layer_info_builder.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+module GeoResources
+  module Discovery
+    class DocumentBuilder
+      class LayerInfoBuilder
+        attr_reader :resource_decorator
+
+        def initialize(resource_decorator)
+          @resource_decorator = resource_decorator
+        end
+
+        # Builds fields about the geospatial file such as geometry and format.
+        # @param [AbstractDocument] discovery document
+        def build(document)
+          document.geom_type = geom_type
+          document.format = format
+        end
+
+        private
+
+          # Gets geo file set decorator
+          # @return [FileSet] geo file set decorator
+          def file_set
+            @file_set ||= begin
+              members = resource_decorator.geo_members
+              members.first.decorate unless members.empty?
+            end
+          end
+
+          # Uses file mime type to determine file format.
+          # @return [String] file format code
+          def format
+            if ControlledVocabulary.for(:geo_image_format).include? geo_mime_type
+              ControlledVocabulary.for(:geo_image_format).find(geo_mime_type).label
+            elsif ControlledVocabulary.for(:geo_raster_format).include? geo_mime_type
+              ControlledVocabulary.for(:geo_raster_format).find(geo_mime_type).label
+            elsif ControlledVocabulary.for(:geo_vector_format).include? geo_mime_type
+              ControlledVocabulary.for(:geo_vector_format).find(geo_mime_type).label
+            end
+          end
+
+          # Returns the 'geo' mime type of the first file attached to the work.
+          # @return [String] file mime type
+          def geo_mime_type
+            return unless file_set
+            file_set.mime_type.first
+          end
+
+          # Uses parent work class to determine file geometry type.
+          # These geom types are used in geoblacklight documents.
+          # @return [String] file geometry type
+          def geom_type
+            case resource_decorator.model_name.to_s
+            when /ScannedMap/
```

Comment:
Why are `RegExp`s used here?  This is not a criticism, but I'm curious as to to why more precise Class-based matching isn't used (using `case resource_decorator.model` and `when ScannedMap`...)

Reply:
This was imported from GeoWorks, so I'm not sure exactly why the class based matching wasn't done. I left it as is because we haven't implemented a RasterWork resource yet, and I want to keep this as placeholder. When we do RasterWorks, it would be a good idea to update this method as you suggest.